### PR TITLE
[NCL-4872] Change REST DTO to use String ids

### PR DESCRIPTION
**Pull request by thescouser89**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by thescouser89** [comment]
Good luck to whoever is going to review this!

**Comment by thescouser89** [comment]
F you unit tests. I'll get you tomorrow

File: dto/src/main/java/org/jboss/pnc/dto/TargetRepository.java [hunk 1]
```diff
@@ -41,8 +41,8 @@
     private final Set<Integer> artifactIds;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private TargetRepository(Set<Integer> artifactIds, Integer id, Boolean temporaryRepo, String identifier, RepositoryType repositoryType, String repositoryPath) {
-        super(id, temporaryRepo, identifier, repositoryType, repositoryPath);
+    private TargetRepository(Set<Integer> artifactIds, String id, Boolean temporaryRepo, String identifier, RepositoryType repositoryType, String repositoryPath) {
+        super(id.toString(), temporaryRepo, identifier, repositoryType, repositoryPath);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/TargetRepository.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/Project.java [hunk 2]
```diff
@@ -44,8 +44,8 @@
     private final List<BuildConfigurationRef> buildConfigurations;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private Project(List<BuildConfigurationRef> buildConfigurations, Integer id, String name, String description, String issueTrackerUrl, String projectUrl) {
-        super(id, name, description, issueTrackerUrl, projectUrl);
+    private Project(List<BuildConfigurationRef> buildConfigurations, String id, String name, String description, String issueTrackerUrl, String projectUrl) {
+        super(id.toString(), name, description, issueTrackerUrl, projectUrl);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/Project.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/ProductVersion.java [hunk 3]
```diff
@@ -63,8 +63,8 @@
     private final List<BuildConfigurationRef> buildConfigurations;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private ProductVersion(ProductRef product, ProductMilestoneRef currentProductMilestone, List<ProductMilestoneRef> productMilestones, List<ProductReleaseRef> productReleases, List<GroupConfigurationRef> groupConfigurations, List<BuildConfigurationRef> buildConfigurations, Integer id, String version, Map<String, String> attributes) {
-        super(id, version, attributes);
+    private ProductVersion(ProductRef product, ProductMilestoneRef currentProductMilestone, List<ProductMilestoneRef> productMilestones, List<ProductReleaseRef> productReleases, List<GroupConfigurationRef> groupConfigurations, List<BuildConfigurationRef> buildConfigurations, String id, String version, Map<String, String> attributes) {
+        super(id.toString(), version, attributes);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/ProductVersion.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/ProductRelease.java [hunk 4]
```diff
@@ -47,8 +47,8 @@
     private final ProductMilestoneRef productMilestone;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private ProductRelease(ProductVersionRef productVersion, ProductMilestoneRef productMilestone, Integer id, String version, SupportLevel supportLevel, Instant releaseDate, String downloadUrl, String issueTrackerUrl) {
-        super(id, version, supportLevel, releaseDate, downloadUrl, issueTrackerUrl);
+    private ProductRelease(ProductVersionRef productVersion, ProductMilestoneRef productMilestone, String id, String version, SupportLevel supportLevel, Instant releaseDate, String downloadUrl, String issueTrackerUrl) {
+        super(id.toString(), version, supportLevel, releaseDate, downloadUrl, issueTrackerUrl);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/ProductRelease.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/ProductMilestone.java [hunk 5]
```diff
@@ -50,8 +50,8 @@
     private final ProductReleaseRef productRelease;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private ProductMilestone(ProductVersionRef productVersion, Set<Integer> performedBuilds, Set<Integer> distributedArtifactIds, ProductReleaseRef productRelease, Integer id, String version, Instant endDate, Instant startingDate, Instant plannedEndDate, String downloadUrl, String issueTrackerUrl) {
-        super(id, version, endDate, startingDate, plannedEndDate, downloadUrl, issueTrackerUrl);
+    private ProductMilestone(ProductVersionRef productVersion, Set<Integer> performedBuilds, Set<Integer> distributedArtifactIds, ProductReleaseRef productRelease, String id, String version, Instant endDate, Instant startingDate, Instant plannedEndDate, String downloadUrl, String issueTrackerUrl) {
+        super(id.toString(), version, endDate, startingDate, plannedEndDate, downloadUrl, issueTrackerUrl);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/ProductMilestone.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/Product.java [hunk 6]
```diff
@@ -47,8 +47,8 @@
     private final Set<ProductVersionRef> productVersions;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private Product(Set<ProductVersionRef> productVersions, Integer id, String name, String description, String abbreviation, String productCode, String pgmSystemName) {
-        super(id, name, description, abbreviation, productCode, pgmSystemName);
+    private Product(Set<ProductVersionRef> productVersions, String id, String name, String description, String abbreviation, String productCode, String pgmSystemName) {
+        super(id.toString(), name, description, abbreviation, productCode, pgmSystemName);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/Product.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/GroupConfiguration.java [hunk 7]
```diff
@@ -51,8 +51,8 @@
     private final List<BuildConfigurationRef> buildConfigurations;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    GroupConfiguration(ProductVersionRef productVersion, List<BuildConfigurationRef> buildConfigurations, Integer id, String name) {
-        super(id, name);
+    GroupConfiguration(ProductVersionRef productVersion, List<BuildConfigurationRef> buildConfigurations, String id, String name) {
+        super(id.toString(), name);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/GroupConfiguration.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/GroupBuild.java [hunk 8]
```diff
@@ -47,8 +47,8 @@
     private final Set<Integer> buildIds;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private GroupBuild(GroupConfigurationRef groupConfig, User user, ProductVersionRef productVersion, Set<Integer> buildIds, Integer id, Instant startTime, Instant endTime, BuildStatus status, Boolean temporaryBuild) {
-        super(id, startTime, endTime, status, temporaryBuild);
+    private GroupBuild(GroupConfigurationRef groupConfig, User user, ProductVersionRef productVersion, Set<Integer> buildIds, String id, Instant startTime, Instant endTime, BuildStatus status, Boolean temporaryBuild) {
+        super(id.toString(), startTime, endTime, status, temporaryBuild);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/GroupBuild.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/BuildConfigurationRevision.java [hunk 9]
```diff
@@ -47,8 +47,8 @@
     private final Map<String, String> genericParameters ;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private BuildConfigurationRevision(SCMRepository repository, ProjectRef project, Environment environment, Map<String, String> genericParameters, Integer id, Integer rev, String name, String description, String buildScript, String scmRevision, Instant creationTime, Instant modificationTime, BuildType buildType) {
-        super(id, rev, name, description, buildScript, scmRevision, creationTime, modificationTime, buildType);
+    private BuildConfigurationRevision(SCMRepository repository, ProjectRef project, Environment environment, Map<String, String> genericParameters, String id, Integer rev, String name, String description, String buildScript, String scmRevision, Instant creationTime, Instant modificationTime, BuildType buildType) {
+        super(id.toString(), rev, name, description, buildScript, scmRevision, creationTime, modificationTime, buildType);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/BuildConfigurationRevision.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/Build.java [hunk 10]
```diff
@@ -66,10 +66,10 @@
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
     private Build(ProjectRef project, SCMRepository repository, Environment environment, Map<String, String> attributes,
             User user, BuildConfigurationRevisionRef buildConfigurationRevision, List<Integer> dependentBuildIds,
-            List<Integer> dependencyBuildIds, Integer id, Instant submitTime, Instant startTime, Instant endTime,
+            List<Integer> dependencyBuildIds, String id, Instant submitTime, Instant startTime, Instant endTime,
             BuildStatus status, String buildContentId, Boolean temporaryBuild, String scmRepositoryURL,
             GroupBuildRef groupBuild) {
-        super(id, submitTime, startTime, endTime, status, buildContentId, temporaryBuild, scmRepositoryURL);
+        super(id.toString(), submitTime, startTime, endTime, status, buildContentId, temporaryBuild, scmRepositoryURL);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/Build.java]: is the toStrong necessary? id should already be String

File: dto/src/main/java/org/jboss/pnc/dto/Artifact.java [hunk 11]
```diff
@@ -46,8 +46,8 @@
     private final Set<Integer> dependantBuildIds;
 
     @lombok.Builder(builderClassName = "Builder", toBuilder = true)
-    private Artifact(TargetRepositoryRef targetRepository, Set<Integer> buildIds, Set<Integer> dependantBuildIds, Integer id, String identifier, ArtifactQuality artifactQuality, String md5, String sha1, String sha256, String filename, String deployPath, Instant importDate, String originUrl, Long size, String deployUrl, String publicUrl) {
-        super(id, identifier, artifactQuality, md5, sha1, sha256, filename, deployPath, importDate, originUrl, size, deployUrl, publicUrl);
+    private Artifact(TargetRepositoryRef targetRepository, Set<Integer> buildIds, Set<Integer> dependantBuildIds, String id, String identifier, ArtifactQuality artifactQuality, String md5, String sha1, String sha256, String filename, String deployPath, Instant importDate, String originUrl, Long size, String deployUrl, String publicUrl) {
+        super(id.toString(), identifier, artifactQuality, md5, sha1, sha256, filename, deployPath, importDate, originUrl, size, deployUrl, publicUrl);
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/Artifact.java]: is the toStrong necessary? id should already be String

**Comment by michalovjan** [comment]
Just a random thought... If we were to add to DTOs Builders a method id(int id) that would convert that int to string and also keep int(String id) at the same time... Maybe it would ease implementation and there won't be that big of a need to correct tests, always convert and such. I hope Lombok has some way to support this. 

WDYT?
@thescouser89 @janinko

**Comment by thescouser89** [comment]
@michalovjan Yeah probably. Will have to fight with annotations to get this working I think?
@janinko Ack! Will double check. That change is so much harder than I anticipated :(

**Comment by michalovjan** [comment]
I've tried to find out how to do that and I am not satisfied with a solution I've found.
Basically, you add a custom method to a builder:
`public static final class Builder {
        public Builder id(Integer id) {
            this.id = String.valueOf(id);
            return this;
        }
    }`

Problem with this is that it overrides the method created by default and you would have to add it manually as another method :/.

` public Builder id(String id) {
            this.id = id;
            return this;
  }`

I don't really like it... Looks awful

**Comment by thescouser89** [comment]
I'll rebase once I figure out why the tests are failing

**Comment by thescouser89** [comment]
sorry @janinko

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
Closing cause it's goign to be hard to fix merge conflicts

**Pull request closed without merging**